Closes #20605: Document variable prefilling via URL parameters

pheus opened the pull request:
<!--
    Thank you for your interest in contributing to NetBox! Please note that
    our contribution policy requires that a feature request or bug report be
    approved and assigned prior to opening a pull request. This helps avoid
    waste time and effort on a proposed change that we might not be able to
    accept.

    IF YOUR PULL REQUEST DOES NOT REFERENCE AN ISSUE WHICH HAS BEEN ASSIGNED
    TO YOU, IT WILL BE CLOSED AUTOMATICALLY.

    Please specify your assigned issue number on the line below.
-->
### Fixes: #20605

<!--
    Please include a summary of the proposed changes below.
-->
Add details on how script form fields can be prefilled using query parameters. Includes examples for string, integer, and object variables, along with considerations for resolving object identifiers.

jnovinger commented on docs/customization/custom-scripts.md:
```diff
@@ -404,6 +404,42 @@ A complete date & time. Returns a `datetime.datetime` object.
+
+*Object variables*
+
+For `ObjectVar`, supply the object’s primary key (PK). For example:
```
For consistency with existing examples in this file, consider using plain text introductions instead of bold/italic pseudo-headings, something like:

> For string and integer variables, when a script defines:
>
> ```python
> from extras.scripts import Script, StringVar, IntegerVar
>
> class MyScript(Script):
>    name = StringVar()
>    count = IntegerVar()
> ```
>
> visiting the URL with matching parameters will prefill the corresponding fields:
>
> ```
> https://<netbox>/extras/scripts/<script_id>/?name=Branch42&count=3
> ```
> For object variables (ObjectVar), supply the object's primary key (PK):
>
> ```
> https://<netbox>/extras/scripts/<script_id>/?device=1
> ```
>

Also, you might consider covering how other Custom Script variables would work or considerations when using:
- `MultiValueDict` behavior: How to pass multiple values for list-type variables
- `ChoiceVar`: Whether choice keys or labels should be used
- `FileVar`: Whether file uploads can be prefilled (not sure we even support this)
- How values with special characters should be represented, URL-encoded?

pheus replied:
Thanks for the thoughtful review and suggestions!

I’ll update the styling. For the other fields, listing everything in prose might get a bit messy. Would a small table with examples be acceptable? If so, I can restructure that section accordingly. Happy to adjust if you have a preferred format.

jnovinger commented:
Derp, meant to leave a comment with that approval, sorry. All in all, this is a great addition as is. Thank you!

I did have a couple of note that I think might make this clearer and follow existing NetBox docs convention.

jnovinger reviewed (commented):
Excellent! I just have one last suggestion. Thanks again for the work on this.

pheus commented on docs/customization/custom-scripts.md:
```diff
@@ -404,6 +404,59 @@ A complete date & time. Returns a `datetime.datetime` object.
+
+Note: The parameter names above are examples; use the actual variable attribute names defined by the script.
+File uploads (`FileVar`) cannot be prefilled via URL parameters.
+
```
That's a great suggestion. Thanks a lot!

pheus committed:
docs(custom-scripts): Document variable prefilling via URL parameters

Add details on how script form fields can be prefilled using query
parameters. Includes examples for string, integer, and object variables,
along with considerations for resolving object identifiers.

Fixes #20605
